src/main/java/org/ice4j/stack/Connector.java

```diff
@@ -207,6 +207,13 @@ else if(localSock.getUDPSocket() != null)
                     return;
 
                 logger.finest("received datagram");
+                logger.finest(String.format("packet - addr: %s port: %d",
+                    packet.getAddress(), packet.getPort()));
+                if (packet.getPort() < 0)
+                {
+                    // force a minimum port of 0 to prevent out of range errors
+                    packet.setPort(0);
```

Comment:
I don't know where the -1 is coming from. I was assuming a faulty peer, so it could be helpful in diagnosing a potential problem. Up to you.